parlai/core/torch_ranker_agent.py

```diff
@@ -0,0 +1,445 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+
+import os
+
+import torch
+from torch import nn
+
+from parlai.core.torch_agent import TorchAgent, Output
+from parlai.core.thread_utils import SharedTable
+from parlai.core.utils import round_sigfigs, padded_3d
+
+
+class TorchRankerAgent(TorchAgent):
+    @staticmethod
+    def add_cmdline_args(argparser):
+        TorchAgent.add_cmdline_args(argparser)
+        agent = argparser.add_argument_group('TorchRankerAgent')
+        agent.add_argument(
+            '-tc', '--train-candidates', type=str, default='batch',
+            choices=['batch', 'inline', 'fixed', 'vocab'],
+            help='The source of candidates during training')
+        agent.add_argument(
+            '-ec', '--eval-candidates', type=str, default='batch',
+            choices=['batch', 'inline', 'fixed', 'vocab'],
+            help='The source of candidates during training')
+        agent.add_argument(
+            '-cands', '--fixed-candidates-path', type=str,
```

Comment:
I suggest the short name for this be `-fcp`, or better yet, there is no short name for this.